### PR TITLE
Refresh notebooks/gufuncs.ipynb

### PR DESCRIPTION
**Pull request by shoyer**
All of the previously commented out examples work now :).
Uncommented these, and added another example showing inserting a new axis.

### CODE REVIEW

**Comment by mattjj** [comment]
Thanks for following up on this! This notebook is a bit of a hidden gem... I'm wondering if we should put `vectorize` in jax.experimental or something...

**Comment by shoyer** [comment]
I'd be up for giving that a try :)

**Comment by mattjj** [comment]
Thanks for this update!

**Review by mattjj — APPROVED** [review]
LGTM!
